src/app/pages/vehicle.cpp

```diff
@@ -137,14 +161,61 @@ QWidget *VehiclePage::dialog_body()
     layout->addWidget(Session::Forge::br(), 1);
     layout->addWidget(this->can_bus_toggle_row(), 1);
 
-    QStringList devices = this->config->get_vehicle_can_bus() ? this->can_devices : this->serial_devices;
+    QStringList devices;
+    switch(config->get_vehicle_can_bus()){
+        //ELM327 USB
+        case ICANBus::VehicleBusType::ELM327USB:
+            devices = this->serial_devices;
+            break;
+        //ELM327 Bluetooth
+        case ICANBus::VehicleBusType::ELM327BT:
+            
+            break;
+        //SocketCAN
+        case ICANBus::VehicleBusType::SocketCAN:
+        default:
+            devices = this->can_devices;
+            break;
+    }
+   
     Selector *interface_selector = new Selector(devices, this->config->get_vehicle_interface(), this->arbiter.forge().font(14), this->arbiter, widget, "disabled");
-    interface_selector->setVisible((this->can_devices.size() > 0) || (this->serial_devices.size() > 0));
-    connect(interface_selector, &Selector::item_changed, [config = this->config](QString item){
-        config->set_vehicle_interface(item);
+    interface_selector->setVisible((this->can_devices.size() > 0) || (this->serial_devices.size() > 0) || (this->paired_bt_devices.size() > 0));
+    connect(interface_selector, &Selector::item_changed, [config = this->config, this](QString item){
```

Comment:
in this lambda capture here, since youre passing in `this` you dont need to capture `config` (since you get get it from `this`)